lib/bundler/cli/add.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+require "bundler/cli/common"
+
+module Bundler
+  class CLI::Add
+    def initialize(options, gem_name)
+      @gem_name = gem_name
+      @options = options
+      @options[:group] = @options[:group].split(",").map(&:strip) if !@options[:group].nil? && !@options[:group].empty?
+    end
+
+    def run
+      version = @options[:version].nil? ? nil : @options[:version].split(",").map(&:strip)
+      dependency = Bundler::Dependency.new(@gem_name, version, @options)
+      Injector.inject([dependency], :conservative_versioning => @options[:version].nil?) # Perform conservative versioning only when version is not specified
```

Comment:
Here you've coupled `bundle add` functionality to `bundle inject` internally. This is ok until we need to diverge the 2 sets of functionalities. I presume the changes you made to `Injector` class does not change the behavior or have any side effects on `bundle inject` and it's designated behavior.

When I wrote the RFC on this feature, it was made clear to me that `bundle inject`'s behavior should remain unchanged.